src/Components/Address/__tests__/useAddressAutocomplete.jest.ts

```diff
@@ -108,18 +115,111 @@ describe("useAddressAutocomplete", () => {
-      )
+      it("does not fetch for a query of less than 5 characters", async () => {
+        const { result } = setupHook({ country: "US" })
+
+        await act(() => {
+          result.current.fetchForAutocomplete({ search: "1234" })
+        })
+
+        expect(mockFetch).not.toHaveBeenCalled()
+      })
+
+      it("returns the correct single-line text for an address without multiple entries", async () => {
+        const { result } = setupHook({ country: "US" })
+
+        act(() => {
+          result.current.fetchForAutocomplete({ search: "401 Broadway" })
+        })
+
+        await waitFor(() => result.current.autocompleteOptions.length > 0)
+
+        const formattedAddress = result.current.autocompleteOptions[0].text
+        expect(formattedAddress).toBe("401 Broadway Fl 25, New York NY 10013")
+      })
+
+      it("returns the correct single-line text for an address with multiple entries", async () => {
+        mockFetch.mockResolvedValue({
+          json: jest.fn().mockResolvedValue({
+            suggestions: [
+              {
+                city: "New York",
+                entries: 2,
+                secondary: "Fl 26",
+                state: "NY",
+                street_line: "402 Broadway",
+                zipcode: "10014",
+              },
+            ],
+          }),
+        })
+
+        const { result } = setupHook({ country: "US" })
+
+        act(() => {
+          result.current.fetchForAutocomplete({ search: "402 Broadway" })
+        })
+
+        await waitFor(() => result.current.autocompleteOptions.length > 0)
+
+        const formattedAddress = result.current.autocompleteOptions[0].text
+        expect(formattedAddress).toBe(
+          "402 Broadway Fl 26 (2 entries), New York NY 10014"
+        )
+      })
+
+      it("resets the suggestions when the country changes", async () => {
+        const { result, rerender } = setupHook({ country: "US" })
+
+        act(() => {
+          result.current.fetchForAutocomplete({ search: "401 Broadway" })
+        })
+
+        await waitFor(() => result.current.autocompleteOptions.length > 0)
+
+        rerender({ country: "UK" })
+
+        expect(result.current.autocompleteOptions).toEqual([])
+      })
+
+      it("resets the suggestions without fetching when the search term is too short", async () => {
+        const { result } = setupHook({ country: "US" })
+
+        act(() => {
+          result.current.fetchForAutocomplete({ search: "401 Broadway" })
+        })
+
+        expect(mockFetch).toHaveBeenCalledTimes(1)
+        await waitFor(() => result.current.autocompleteOptions.length > 0)
+
+        mockFetch.mockClear()
+
+        act(() => {
+          result.current.fetchForAutocomplete({ search: "40" })
+        })
+
+        expect(result.current.autocompleteOptions).toEqual([])
+        expect(mockFetch).not.toHaveBeenCalled()
+      })
+
+      it.skip("debounces calls to Smarty API", async () => {
```

Comment:
This test looks good, but it's being skipped so I'm not sure if it passes or not.

Reply:
Testing the debouncing is good to do for this hook for sure, but it actually isn't passing unfortunately :( 

Also, I have it as skipped because there was a follow up conversation w/ design about reconsidering how the UI handles the key inputs so this test is TBD. @erikdstock wrote another ticket as a follow up here: https://artsyproduct.atlassian.net/browse/EMI-1459

Reply:
Got it. I was missing out on that context 👍

Reply:
I agree, but the existing set up for adding fake jest timers may be useful for updating the tests once someone gets around to it? No opinion either way though!

Reply:
Maybe it makes sense to remove rather than skip for now, but then you could immediately begin on the debounce ticket and just paste it right back in?